src/main/java/sirius/biz/storage/layer3/VFSController.java

```diff
@@ -270,14 +270,42 @@ public void listAPI(WebContext ctx, JSONStructuredOutput out) {
         }
         parent.assertExistingDirectory();
 
-        FileSearch search = FileSearch.iterateAll(child -> outputFile(out, child.name(), child));
+        outputPath(out, parent);
+        outputChildren(ctx, out, parent);
+
+        out.property("canCreateChildren", parent.canCreateChildren());
+    }
+
+    private void outputChildren(WebContext ctx, JSONStructuredOutput out, VirtualFile parent) {
+        out.beginArray("children");
 
+        FileSearch search = FileSearch.iterateAll(child -> outputFile(out, child.name(), child));
         if (ctx.get("onlyDirectories").asBoolean()) {
             search.withOnlyDirectories();
         }
-
         search.withPrefixFilter(ctx.get("filter").asString());
-        search.withLimit(new Limit(0, 100));
+
+        // because of the additional ".."-entry for the parent, we need to adjust the pagination skip/limit
+        boolean hasParent = parent.parent() != null;
+        int skip = ctx.get("skip").asInt(0);
+        Integer limit = ctx.get("maxItems").getInteger();
+        if (hasParent) {
+            if (skip > 0) {
+                skip--;
+            } else {
+                outputFile(out, "..", parent.parent());
```

Comment:
Should we extract the ".." as a constant to at least give it a descriptive name?